src/mean_function.jl

```diff
@@ -1,5 +1,20 @@
+"""
+`MeanFunction` introduces an API for treating the prior mean function appropriately.
+On the abstract level, all `MeanFunction` are functions.
+However we generally want to evaluate them on a collection of inputs.
+To this effect, we provide the `mean_vector(::MeanFunction, ::AbstractVector)`
+function, which is equivalent to `map` but with possibilities of optimizations
+(for [`ZeroMean`](@ref) and [`ConstMean`](@ref) for example).
+"""
 abstract type MeanFunction end
 
+"""
+    mean_vector(m::MeanFunction, x::AbstractVector)::AbstractVector
```

Comment:
```suggestion
    mean_vector(m::MeanFunction, x::AbstractVector)::AbstractVector{<:Real}
```
We can be sure that a mean function outputs a vector of reals